### PR TITLE
fix: type errors

### PR DESCRIPTION
**Pull request by GiveMe-A-Name**
## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/builder/uni-builder/package.json [hunk 1]
```diff
@@ -76,6 +76,7 @@
   },
   "devDependencies": {
     "@modern-js/builder-plugin-node-polyfill": "workspace:*",
+    "@modern-js/server-core": "workspace:*",
```
Comment (yimingjfe) [inline packages/builder/uni-builder/package.json:79]: Why was this dependency included?

File: packages/server/core/package.json [hunk 2]
```diff
@@ -31,7 +31,7 @@
       "default": "./dist/cjs/index.js"
     },
     "./base": {
-      "types": "./base/index.ts",
+      "types": "./dist/types/base/index.d.ts",
```
Comment (yimingjfe) [inline packages/server/core/package.json:34]: It should not be modified here，please see publishConfig

Comment (GiveMe-A-Name) [reply packages/server/core/package.json:34]: Is just for source-code compiler?

File: packages/builder/uni-builder/package.json (hunk 1 above)
Comment (GiveMe-A-Name) [reply packages/builder/uni-builder/package.json:79]: uni-builder need ServerBaseOptions type from modern-js/server-core

File: packages/server/core/package.json (hunk 2 above)
Comment (GiveMe-A-Name) [reply packages/server/core/package.json:34]: uni-builder use tsc to compiler, it seems not support source-code compile.

**Review by yimingjfe — APPROVED** [review]

**Pull request merged**